core/src/main/scala/com/nvidia/spark/rapids/tool/tuning/TuningEntryTrait.scala

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.tool.tuning
+
+import scala.collection.mutable.ListBuffer
+
+import com.nvidia.spark.rapids.tool.tuning.TuningOpTypes.TuningOpType
+
+import org.apache.spark.sql.rapids.tool.util.StringUtils
+
+/**
+ * A trait that defines the behavior of the Tuning Entry.
+ */
+trait TuningEntryTrait {
+  val name: String
+  // The value recommended by the AutoTuner
+  var tunedValue: Option[String]
+  // The original value of the property from the event log
+  var originalValue: Option[String]
+  var enabled: Boolean = true
+
+  // The type of tuning operation to be performed
+  var tuningOpType: TuningOpType = TuningOpTypes.UNKNOWN
+  // Comments specific to the property
+  val comments: ListBuffer[String] = ListBuffer[String]()
+  // The value to be used when the property is unresolved.
+  var fillUnresolved: Option[String] = Some("[FILL_IN_VALUE]")
+
+  def toConfString: String = {
+    "--conf %s=%s".format(name, getTuneValue())
+  }
+
+  def setOriginalValueFromDefaultSpark(): Unit
+
+  /**
+   * Indicates if the property is resolved by the AutoTuner or not. This is used to distinguish
+   * the properties that were not tuned due to a failure in the AutoTuner. For example,
+   * shuffleManager was not able to find the className but we still want to include this
+   * property in the bootstrap configuration.
+   * @param fill the value to be used to fill in the gap if necessary.
+   */
+  def markAsUnresolved(fill: Option[String] = None): Unit = {
+    setTuningOpType(TuningOpTypes.UNRESOLVED)
+    if (fill.isDefined) {
+      fillUnresolved = fill
+    }
+  }
+
+  /**
+   * Disables the property. This is used to turnoff the property if it is not applicable.
+   */
+  def disable(): Unit = {
+    enabled = false
+  }
+
+  /**
+   * Returns the value of the property as a string.
+   * When the tunedValue is not set, it will set it as [Fill_IN_VALUE] so that it can be replaced
+   * by the user. This is used because the AutoTuner may not be able to successfully make
+   * recommendations. Yet, we want to include that in the final tuning report.
+   * Note that the name is not "getTunedValue" because the purpose is different.
```

Comment:
QQ: does this refer to the function name `getTuneValue`, which is not `getTunedValue`?
Could you help elaborate on the differentiation here?

Reply:
There is a field called `tunedValue`.
A method called `getTunedValue()` in Java standard is a `getter` which is expected to return the field named `TunedValue`.

The implemented method here is named `getTuneValue` because it does not necessarily represent the actual value in the field:

-  Instead of `option[string]`, it returns a `string`.
-  It processes the field and may replace with a predefined pattern.

Reply:
understood, thanks @amahussein!